vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/log/LogcatMessageParser.kt

```diff
@@ -0,0 +1,145 @@
+package com.malinskiy.marathon.android.adam.log
+
+import com.malinskiy.marathon.android.AndroidDevice
+import java.util.regex.Pattern
+
+/**
+ * Class to parse raw output of { @code adb logcat -v long } to { @link LogCatMessage } objects.
+ */
+open class LogCatMessageParser {
+    var previousHeader: LogCatHeader? = null
+
+    /**
+     * Parse a header line into a [LogCatHeader] object, or `null` if the input line
+     * doesn't match the expected format.
+     *
+     * @param line   raw text that should be the header line from logcat -v long
+     * @param device device from which these log messages have been received
+     * @return a [LogCatHeader] which represents the passed in text
+     */
+    open fun processLogHeader(line: String, device: AndroidDevice): LogCatHeader? {
+        val matcher = HEADER.matcher(line)
+        if (!matcher.matches()) {
+            return null
+        }
+        val dateTime: LogCatTimestamp = LogCatTimestamp.fromString(matcher.group(1))
+        val processId = parseProcessId(matcher.group(2))
+        val threadId = parseThreadId(matcher.group(3))
+        val priority: Log.LogLevel = parsePriority(matcher.group(4))
+        val tag = matcher.group(5)
+        previousHeader = LogCatHeader(
```

Comment:
why not return it here?

Reply:
we need to save the header for all subsequent log lines